nsis/src/nsis/file/NsisExecutable.java

```diff
@@ -0,0 +1,58 @@
+package nsis.file;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import generic.continues.GenericFactory;
+import ghidra.app.util.bin.ByteProvider;
+import ghidra.app.util.bin.format.FactoryBundledWithBinaryReader;
+import ghidra.app.util.bin.format.pe.PortableExecutable.SectionLayout;
+
+public class NsisExecutable {
+
+	public static final String NAME = "NULLSOFT_SCRIPTABLE_INSTALLER_SYSTEM";
+
+	private FactoryBundledWithBinaryReader reader;
+	private long header_offset;
+
+	public NsisExecutable() {
+	}
+
+	public static NsisExecutable createNsisExecutable(GenericFactory factory,
+			ByteProvider bp, SectionLayout layout) {
+		NsisExecutable nsisExecutable = (NsisExecutable) factory
+				.create(NsisExecutable.class);
+		nsisExecutable.initNsisExecutable(factory, bp, layout);
+		return nsisExecutable;
+	}
+
+	private void initNsisExecutable(GenericFactory factory, ByteProvider bp,
+			SectionLayout layout) {
+		this.reader = new FactoryBundledWithBinaryReader(factory, bp, true);
+		this.header_offset = -1;
+		findHeaderOffset();
+		// TODO Init sections and headers
+	}
+
+	private void findHeaderOffset() {
```

Comment:
This function should be declared as "throws IOException".

In general we want to avoid catching exceptions unless we can deal with them properly. In this case the exception would be related to some kind of IO error, like a file system error. The error should be thrown so that code higher up in the call stack could potentially deal with it like showing an error message to the user.

You will have to mark other functions as "throws IOException" as well.

You can see that when Ghidra processes PE files it also declares the functions as throws IOException. 
https://github.com/dragonGR/Ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/util/opinion/PeLoader.java#L98

https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/pe/PortableExecutable.java#L81

Reply:
Good point. I updated it to throw appropriate exceptions!